packages/office-ui-fabric-react/src/components/HoverCard/HoverCard.base.tsx

```diff
@@ -82,6 +83,16 @@ export class HoverCardBase extends BaseComponent<IHoverCardProps, IHoverCardStat
     }
   }
 
+  public dismiss = (): void => {
+    this._async.clearTimeout(this._openTimerId);
+    this._async.clearTimeout(this._dismissTimerId);
+    this.setState({
+      isHoverCardVisible: false,
+      mode: ExpandingCardMode.compact,
+      openMode: OpenCardMode.hover
+    });
```

Comment:
Hmm, what happens if the component is not mounted? I'm not sure what our convention is around dismiss implementations (if we have one.) Also, David has been getting rid of BaseComponent which would affect this function.

Reply:
We have one implementation of a public `dismiss` method in Panel. Also, I updated the code slightly so you might wanna take another look when you find some time. As to the question what happens if the component is not mounted: not sure what are your concerns here...

Reply:
I don't think it will cause an exception based on BaseComponent's implementation, but it does seem like it will cause a leak by instantiating a new internal _async that is never cleaned up. Additionally, calling setState on an unmounted component will generate a React warning. Ideally this function should not be no-op if the component is not mounted.

Reply:
How would you recommend making sure the component is mounted? One way I've seen this implemented is in Layer with a state property.

Reply:
I don't think we have to solve it as part of this PR, but I think making "safe" imperative functions (like `focus`) is something we could create a convention for using Hooks.